### PR TITLE
Add usage documentation and update README

### PR DESCRIPTION
**Pull request by Mordil**

### CODE REVIEW

File: docs/Usage.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+// USGas(3.279), USGas(3.269)
+```
+
+### Currency Protocols
+[[Back to Top]](#table-of-contents)
+
+**SwiftCurrency** provides two protocols for working with currencies:
+
+1. [`AnyCurrency`][AnyCurrency]
+    -  The basic representation of a currency, which be used as an [existential value][Existentials] of a `CurrencyProtocol` instance that can be passed around in place of a concrete type such as `USD`
```
Comment (kaishin) [inline docs/Usage.md]: which _can_ be used, I guess?

File: docs/Usage.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+    - [There are limitations to using this protocol](#api-limitations)
+1. [`CurrencyProtocol`][CurrencyProtocol]
+    - Refines `AnyCurrency`, but with `Self` requirements as it also refines Standard Library protocols
+such as [`Comparable`](https://developer.apple.com/documentation/swift/comparable) and [`Hashable`](https://developer.apple.com/documentation/swift/hashable)
+
+Guidelines for when to use `AnyCurrency` vs `CurrencyProtocol`:
+
+1. When creating a [custom currency](#custom-currencies), conform to `CurrencyProtocol`.
+1. If you need to extend _any_ currency type - use `AnyCurrency`.
+1. If you are extending types with associatedtypes that are a Currency, such as `Array` or `Set`, use `AnyCurrency`.
```
Comment (kaishin) [inline docs/Usage.md]: associated types.

**Review by kaishin — COMMENTED** [review]
One typo and minor nit 😉

File: docs/Usage.md (hunk 1 above)
Comment (Mordil) [reply docs/Usage.md]: Oops. Good catch

**Pull request merged**